src/zjs_aio.c

```diff
@@ -193,66 +131,59 @@ static ZJS_DECL_FUNC(zjs_aio_pin_read_async)
 
 static ZJS_DECL_FUNC(zjs_aio_pin_close)
 {
-    u32_t pin;
-    zjs_obj_get_uint32(this, "pin", &pin);
+    FTRACE_JSAPI;
     GET_AIO_HANDLE(this, handle);
-
     aio_free_cb(handle);
-
     return ZJS_UNDEFINED;
 }
 
 static ZJS_DECL_FUNC(zjs_aio_open)
 {
-    // args: initialization object
-    ZJS_VALIDATE_ARGS(Z_OBJECT);
-
-    jerry_value_t data = argv[0];
-
-    u32_t pin;
-    if (!zjs_obj_get_uint32(data, "pin", &pin))
-        return zjs_error("missing required field (pin)");
+    FTRACE_JSAPI;
+    // args: initialization object or int/string pin number
+    ZJS_VALIDATE_ARGS(Z_NUMBER Z_STRING Z_OBJECT);
+
+    ZVAL_MUTABLE pin_str = 0;
+    jerry_value_t pin_val = argv[0];
+    jerry_value_t init = 0;
+
+    if (jerry_value_is_object(argv[0])) {
+        init = argv[0];
+        pin_str = zjs_get_property(init, "pin");
+        pin_val = pin_str;
+    }
 
-    if (pin < AIO_MIN || pin > AIO_MAX) {
-        DBG_PRINT("PIN: #%u\n", pin);
-        return zjs_error("pin out of range");
+    char devname[20];
+    int pin = zjs_board_find_aio(pin_val, devname, 20);
+    if (pin == FIND_PIN_INVALID) {
+        return TYPE_ERROR("bad pin argument");
+    }
+    else if (pin == FIND_DEVICE_FAILURE) {
+        return zjs_error("device not found");
```

Comment:
This string appears 3 times, should probably be a #define or something similar.

Reply:
Yeah, a system for doing that consistently would be nice, but as long as it's an exact match for the string it's not really a problem... the compiler notices they're identical and uses the same static string.